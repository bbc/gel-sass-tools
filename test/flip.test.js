describe('flip()', () => {
  describe('in ltr mode', () => {
    it('is maintained when setting text alignment', async () => {
      const css = await global.compileCSS('test/data/flip/text-align.scss');

      expect(css).toMatchSnapshot();
    });

    it('is maintained when setting float', async () => {
      const css = await global.compileCSS('test/data/flip/float.scss');

      expect(css).toMatchSnapshot();
    });

    it('is maintained when setting direction', async () => {
      const css = await global.compileCSS('test/data/flip/direction.scss');

      expect(css).toMatchSnapshot();
    });

    it('is maintained when setting flex-direction', async () => {
      const css = await global.compileCSS('test/data/flip/flex-direction.scss');

      expect(css).toMatchSnapshot();
    });
  });

  describe('in rtl mode', () => {
    it('is applied when setting text alignment', async () => {
      const css = await global.compileCSS('test/data/flip/text-align--rtl.scss');

      expect(css).toMatchSnapshot();
    });

    it('is applied when setting float', async () => {
      const css = await global.compileCSS('test/data/flip/float--rtl.scss');

      expect(css).toMatchSnapshot();
    });

    it('is applied when setting direction', async () => {
      const css = await global.compileCSS('test/data/flip/direction--rtl.scss');

      expect(css).toMatchSnapshot();
    });

    it('is applied when setting flex-direction', async () => {
      const css = await global.compileCSS('test/data/flip/flex-direction--rtl.scss');

      expect(css).toMatchSnapshot();
    });
  });
});
