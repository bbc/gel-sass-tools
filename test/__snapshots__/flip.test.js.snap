// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flip() in ltr mode is maintained when setting direction 1`] = `
".demo {
  direction: ltr;
}"
`;

exports[`flip() in ltr mode is maintained when setting flex-direction 1`] = `
".demo {
  flex-direction: row-reverse;
}"
`;

exports[`flip() in ltr mode is maintained when setting float 1`] = `
".demo {
  float: left;
}"
`;

exports[`flip() in ltr mode is maintained when setting text alignment 1`] = `
".demo {
  text-align: left;
}"
`;

exports[`flip() in rtl mode is applied when setting direction 1`] = `
".demo {
  direction: rtl;
}"
`;

exports[`flip() in rtl mode is applied when setting flex-direction 1`] = `
".demo {
  flex-direction: row;
}"
`;

exports[`flip() in rtl mode is applied when setting float 1`] = `
".demo {
  float: right;
}"
`;

exports[`flip() in rtl mode is applied when setting text alignment 1`] = `
".demo {
  text-align: right;
}"
`;
